core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -247,7 +247,7 @@ abstract class AbstractFetcherThread(name: String,
         val highWatermark = partitionState.fetchOffset
         val truncationState = OffsetTruncationState(highWatermark, truncationCompleted = true)
 
-        info(s"Truncating partition $tp to local high watermark $highWatermark")
+        info(s"Truncating partition $tp to $truncationState due to local high watermark $highWatermark")
```

Comment:
nit: the `toString` for `OffsetTruncationState` is a little unconventional. Maybe we could replace it with something like this:
```scala
  override def toString: String = s"TruncationState(offset=$offset, completed:$truncationCompleted)"
```
Then we can change this from "to" to "with" so the message reads (e.g.):
> Truncating partition foo-0 with TruncationState(offset=19, completed=true) based on local high watermark 19.

Reply:
Done. Updated log messages:
```
Truncating partition topic1-0 with TruncationState(offset=0, completed=true) due to local high watermark 0
Truncating partition topic1-0 with TruncationState(offset=5, completed=true) due to leader epoch and offset EpochEndOffset(errorCode=0, partition=0, leaderEpoch=0, endOffset=5)
```